includes/AMP/Sanitization.php

```diff
@@ -0,0 +1,459 @@
+				// mode if the user is actually logged in. This prevents the dev mode from being served to crawlers
+				// when they index the AMP version.
+				( is_admin_bar_showing() && is_user_logged_in() )
+			)
+		);
+	}
+
+	/**
+	 * Returns a list of sanitizers to use.
+	 *
+	 * This is replica of amp_get_content_sanitizers() to avoid
+	 * loading amp-helper-functions.php due to side-effects like
+	 * accessing options from the database, requiring AMP__VERSION,
+	 * and causing conflicts with our own amp_is_request() compat shim.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @return array Sanitizers.
+	 */
+	protected function get_sanitizers() {
+		$sanitizers = [
+			AMP_Script_Sanitizer::class            => [],
+			AMP_Style_Sanitizer::class             => [
+
+				/*
+				 * @todo Enable by default and allow filtering once AMP_Style_Sanitizer does not call AMP_Options_Manager
+				 *       which in turn requires AMP__VERSION to be defined.
+				 */
+				'allow_transient_caching' => false,
+				'use_document_element'    => true,
+			],
+			Meta_Sanitizer::class                  => [],
+			AMP_Layout_Sanitizer::class            => [],
+			Canonical_Sanitizer::class             => [],
+			Story_Sanitizer::class                 => [],
+			AMP_Tag_And_Attribute_Sanitizer::class => [],
+		];
+
+		/**
+		 * Filters the content sanitizers.
+		 *
+		 * @since 1.1.0
+		 *
+		 * @param array  $sanitizers Sanitizers.
+		 */
+		$sanitizers = apply_filters( 'web_stories_amp_sanitizers', $sanitizers );
+
+		if ( $this->is_amp_dev_mode() ) {
+			/**
+			 * Filters the XPath queries for elements that should be enabled for dev mode.
+			 *
+			 * By supplying XPath queries to this filter, the data-ampdevmode attribute will automatically be added to the
+			 * root HTML element as well as to any elements that match the expressions. The attribute is added to the
+			 * elements prior to running any of the sanitizers.
+			 *
+			 * @since 1.3
+			 * @param string[] $element_xpaths XPath element queries. Context is the root element.
+			 */
+			$dev_mode_xpaths = (array) apply_filters( 'web_stories_amp_dev_mode_element_xpaths', [] );
+
+			if ( is_admin_bar_showing() ) {
+				$dev_mode_xpaths[] = '//*[ @id = "wpadminbar" ]';
+				$dev_mode_xpaths[] = '//*[ @id = "wpadminbar" ]//*';
+				$dev_mode_xpaths[] = '//style[ @id = "admin-bar-inline-css" ]';
+			}
+
+			$sanitizers = array_merge(
+				[
+					\AMP_Dev_Mode_Sanitizer::class => [
+						'element_xpaths' => $dev_mode_xpaths,
+					],
+				],
+				$sanitizers
+			);
+		}
+
+		// Force certain sanitizers to be at end.
+		// AMP_Style_Sanitizer needs to catch any CSS changes from previous sanitizers.
+		// AMP_Tag_And_Attribute_Sanitizer must come at the end to clean up any remaining issues the other sanitizers didn't catch.
+		foreach ( [ AMP_Style_Sanitizer::class, Meta_Sanitizer::class, AMP_Tag_And_Attribute_Sanitizer::class ] as $class_name ) {
```

Comment:
Recent change introduced in https://github.com/ampproject/amp-wp/pull/5366:

```suggestion
		foreach ( [ AMP_Layout_Sanitizer::class, AMP_Style_Sanitizer::class, Meta_Sanitizer::class, AMP_Tag_And_Attribute_Sanitizer::class ] as $class_name ) {
```